### PR TITLE
[14.0][ADD] stock_location_package_restriction

### PR DESCRIPTION
**Pull request by jbaudoux**
Control if the location must contain only (one) package(s)

Complementary to https://github.com/OCA/stock-logistics-warehouse/tree/16.0/stock_location_product_restriction

cc @rousseldenis @lmignon

### CODE REVIEW

**Review by JuMiSanAr — APPROVED** [review]

File: stock_location_package_restriction/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Raumschmiede (http://www.raumschmiede.de)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock Location Package Restriction",
+    "summary": """
+        Control if the location can contain products not on a package""",
```
Comment (sebalix) [inline stock_location_package_restriction/__manifest__.py]:
```suggestion
        Control if the location can contain products not in a package""",
```

File: stock_location_package_restriction/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+NORESTRICTION = "norestriction"
+
+
+class StockLocation(models.Model):
+    _inherit = "stock.location"
+
+    package_restriction = fields.Selection(
+        selection=lambda self: self._selection_package_restriction(),
+        help="""
+            Control if the location can contain products not on a package.
```
Comment (sebalix) [inline stock_location_package_restriction/models/stock_location.py]:
```suggestion
            Control if the location can contain products not in a package.
```

**Review by sebalix — APPROVED** [review]
LG

File: stock_location_package_restriction/__manifest__.py (hunk 1 above)
Comment (jbaudoux) [reply stock_location_package_restriction/__manifest__.py]: fixed

File: stock_location_package_restriction/models/stock_location.py (hunk 2 above)
Comment (jbaudoux) [reply stock_location_package_restriction/models/stock_location.py]: fixed

**Pull request merged**